docs/reference-guide/modules/commands/pages/command-dispatchers.adoc

```diff
@@ -17,154 +17,421 @@ As such, it is the infrastructure component that is aware of which component can
+// Send command and get CommandResult
+CommandResult result = dispatcher.send(command);
+
+// Send command with metadata
+CommandResult result = dispatcher.send(command, metadata);
+
+// Send command and get CompletableFuture with expected type
+CompletableFuture<String> future = dispatcher.send(command, String.class);
+
+// Attach handlers to the result
+dispatcher.send(command)
+         .onSuccess(String.class, cardNumber -> {
+             // Handle success
+         })
+         .onError(exception -> {
+             // Handle error
+         });
+----
+
+Notice that the `CommandDispatcher` API does *not* have a `ProcessingContext` parameter.
+This is because the `CommandDispatcher` is created for a specific `ProcessingContext` and automatically uses it for all dispatched commands.
+
+[TIP]
+====
+Use the `CommandDispatcher` whenever you dispatch commands from within a message handler.
+Use the `CommandGateway` (with `null` for `ProcessingContext`) when dispatching from outside a handler (for example, from an HTTP endpoint).
+====
 
 == Command dispatching results
 
-Dispatching commands will, generally speaking, have two possible outcomes:
+Dispatching commands will generally have two possible outcomes:
 
 . Command handled successfully, and
 . command handled exceptionally
 
-The outcome to some extent depends on the dispatching process, but more so on the implementation of the command handler.
-Thus, if the `@CommandHandler` annotated function throws an exception due to some business logic, it will be that exception which will be the result of dispatching the command.
+The outcome depends on the implementation of the command handler.
+If the `@CommandHandler` annotated function throws an exception due to some business logic, it will be that exception which will be the result of dispatching the command.
+
+=== Return values from command handlers
+
+Command handlers can return values, and the framework will propagate these through the `CommandResultMessage` (or directly through `CommandResult`/`CompletableFuture` depending on the API used).
+However, it's important to understand the intent behind commands versus queries:
+
+* **Commands** express an intent to change application state.
+The primary goal of dispatching a command is to execute that state change, not to retrieve information.
+* **Queries** are designed specifically for retrieving information from the application.
+
+If the primary purpose of sending a message is to obtain a result value, consider whether it should be designed as a xref:queries:index.adoc[query message] instead of a command.
+
+That said, there are legitimate cases where command handlers return values:
+
+* **Entity identifiers**: Returning the identifier of a newly created aggregate or entity is a common pattern.
+The framework has built-in support for this: when a `@CommandHandler` annotated constructor executes successfully, the value of the `@AggregateIdentifier` annotated field is automatically returned.
```

Comment:
We do not have constructor command handlers anymore, only static factory methods.
```suggestion
The framework has built-in support for this: when a `@CommandHandler` annotated static handler executes successfully, the value of the `@AggregateIdentifier` annotated field is automatically returned.
```